libdnf5/conf/vars.cpp

```diff
@@ -223,17 +223,52 @@ std::string Vars::substitute(const std::string & text) const {
     return res;
 }
 
+std::tuple<std::string, std::string> Vars::split_releasever(const std::string & releasever) {
+    // Uses the same logic as splitReleaseverTo in libzypp
+    std::string releasever_major;
+    std::string releasever_minor;
+    const auto pos = releasever.find('.');
+    if (pos == std::string::npos) {
+        releasever_major = releasever;
+    } else {
+        releasever_major = releasever.substr(0, pos);
+        releasever_minor = releasever.substr(pos + 1);
+    }
+    return std::make_tuple(releasever_major, releasever_minor);
+}
+
 void Vars::set(const std::string & name, const std::string & value, Priority prio) {
     auto it = variables.find(name);
-    if (it != variables.end()) {
-        if (it->second.priority > prio) {
-            return;
-        }
+
+    // Do nothing if the var is already set with a higher priority
+    if (it != variables.end() && prio < it->second.priority) {
+        return;
+    }
+
+    // Whenever releasever is set, split it into major and minor parts
+    if (name == "releasever") {
+        const auto [releasever_major, releasever_minor] = split_releasever(value);
+        set("releasever_major", releasever_major, prio);
+        set("releasever_minor", releasever_minor, prio);
+    }
```

Comment:
I am not sure whether automatic detection during the setting is the right approach. Anyway it is not wrong.

I have some concerns to priority and setting empty values.
The current code using the same priority for derived variables. Why cannot we use a lover priority? Example user define from commandline `releasever_major` and then `releasever` but it overrides `releasever_major` because autodetected value has the same priority.

With empty value I guess that we should not substitute `$releasever_minor` when `releasever` = f38. DNF does that in other places (when no releasever (in installroot), it does not substitute it) and I somehow feel it  would be good to keep the same pattern.  This behavior is different from shell, but it helps with debugging.

Reply:
Hmm, playing with priorities and setting `releasever_major` with artificially lowered priority doesn't seem right to me. I believe the root issue here lies in the design itself - `releasever_major` and `releasever_minor` are dependent on the value of the `releasever` variable and should be treated as kind of  "readonly" (I'd even consider disallowing independent setting, allowing changes only through `releasever`).

Reply:
That would be possible, but it would require to implement locking mechanism for Vars, like we have for configuration and it will create an exception in the logic.

Reply:
> Hmm, playing with priorities and setting `releasever_major` with artificially lowered priority doesn't seem right to me. I believe the root issue here lies in the design itself - `releasever_major` and `releasever_minor` are dependent on the value of the `releasever` variable and should be treated as kind of "readonly" (I'd even consider disallowing independent setting, allowing changes only through `releasever`).

I agree, I think this is the better approach, I made `releasever_major` and `releasever_minor` read-only. Let me know what you think.

Reply:
I think this sufficiently addressed this feedback.

Reply:
I think this sufficiently addresses the feedback. And this is the only feedback blocking me from merging this.

@j-mracek are we good with this now?